examples/cosmos/lightclient/Lightclient.qnt

```diff
@@ -0,0 +1,550 @@
+                        state' =
+                            if (current.header.height < TARGET_HEIGHT) "working"
+                            else "finishedSuccess",
+                        nondet newHeight = oneOf(HEIGHTS)
+                        all {
+                            CanScheduleTo(newHeight, current, nextHeight, TARGET_HEIGHT),
+                            nextHeight' = newHeight
+                        }
+                    }
+                else if (verdict == "NOT_ENOUGH_TRUST")
+                    /*
+                    do nothing: the light block current passed validation, but the validator
+                    set is too different to verify it. We keep the state of
+                    current at StateUnverified. For a later iteration, Schedule
+                    might decide to try verification of that light block again.
+                    */
+                    all {
+                        lightBlockStatus' = LightStoreUpdateStates(lightBlockStatus, nextHeight, "StateUnverified"),
+                        latestVerified' = latestVerified,
+                        state' = state,
+                        nondet newHeight = oneOf(HEIGHTS)
+                        all {
+                            CanScheduleTo(newHeight, latestVerified, nextHeight, TARGET_HEIGHT),
+                            nextHeight' = newHeight 
+                        }
+                    }
+                else
+                    // verdict is some error code
+                    all {
+                        lightBlockStatus' = LightStoreUpdateStates(lightBlockStatus, nextHeight, "StateFailed"),
+                        state' = "finishedFailure",
+                        latestVerified' = latestVerified,
+                        nextHeight' = nextHeight
+                    }
+            }   
+        }
+    }
+
+    // The terminating condition of VerifyToTarget.
+    //
+    // [LCV-FUNC-MAIN.1::TLA-LOOPCOND.1]
+    action VerifyToTargetDone: bool = all {
+        latestVerified.header.height >= TARGET_HEIGHT,
+        state' = "finishedSuccess",
+        nextHeight' = nextHeight,
+        nprobes' = nprobes,
+        fetchedLightBlocks' = fetchedLightBlocks,
+        lightBlockStatus' = lightBlockStatus,
+        latestVerified' = latestVerified,
+        prevVerified' = prevVerified,
+        prevCurrent' = prevCurrent,
+        prevLocalClock' = prevLocalClock,
+        prevVerdict' = prevVerdict
+    }
+        
+
+    /*
+    The local and global clocks can be updated. They can also drift from each other.
+    Note that the local clock can actually go backwards in time.
+    However, it still stays in the drift envelope
+    of [refClock - REAL_CLOCK_DRIFT, refClock + REAL_CLOCK_DRIFT].
+    */
+    action AdvanceClocks: bool = {
+        nondet refClockTm = oneOf(Int)
+        nondet tm = oneOf(Int)
+        all {
+            BC::AdvanceTimeTo(refClockTm),
+            tm >= 0,
```

Comment:
Also here, it would help the simulator, when we choose `tm` from `Nat`, not `Int`.